agents/packages/ethlisbon/skills/price_prophet/ml_tools.py

```diff
@@ -126,7 +127,8 @@ def train_model(x: pd.DataFrame, random_state: int):
     """Train model"""
 
     # currently we interpolate and discard features with NaN
-    x_imputed = impute(x)
+    x_imputed = x.dropna(axis=1, how="all").dropna()
+    x_imputed.reset_index(drop=True, inplace=True)  # for forecaster
```

Comment:
first we drop all NaN columns, then the NaN containing rows (any). We have many starting with NaN (e.g. moving average filters) which we were discarding entirely before (only using interpolate and dropna(axis=1))